trunk/SUAVE/Analyses/Atmospheric/US_Standard_1976.py

```diff
@@ -0,0 +1,172 @@
+            print 'Warning: US Standard Atmosphere being used outside expected conditions'
+
+        # convert input if necessary
+        if isinstance(zs, int): 
+            zs = np.array([float(zs)])
+        elif isinstance(zs, float):
+            zs = np.array([zs])
+
+        # convert geometric to geopotential altitude
+        zs = zs/(1 + zs/Rad)
+        
+        # Remove redudant brackets
+        if len(zs.shape) == 2:
+            zs = zs[:,0]
+
+        # initialize return data
+        p = np.zeros(np.size(zs))
+        T = np.zeros(np.size(zs))
+        rho = np.zeros(np.size(zs))
+        a = np.zeros(np.size(zs))
+        mew = np.zeros(np.size(zs))
+        z0 = np.zeros(np.size(zs))
+        T0 = np.zeros(np.size(zs))
+        p0 = np.zeros(np.size(zs))
+        alpha = np.zeros(np.size(zs))
+        
+        # evaluate at each altitude
+        zmin = self.breaks.altitude[0]
+        zmax = self.breaks.altitude[-1]
+        
+        if np.amin(zs) < zmin:
+            print "Warning: altitude requested below minimum for this atmospheric model; returning values for h = -2.0 km"
+            zs[zs < zmin] = zmin
+        if np.amax(zs) > zmax:
+            print "Warning: altitude requested above maximum for this atmospheric model; returning values for h = 86.0 km"   
+            zs[zs > zmax] = zmax
+        for i in range(len(self.breaks.altitude)-1): # this uses >= and <= to capture both edges and because values should be the same at the edges
+            z0[(zs>=self.breaks.altitude[i]) & (zs <= self.breaks.altitude[i+1])] = self.breaks.altitude[i]
+            T0[(zs>=self.breaks.altitude[i]) & (zs <= self.breaks.altitude[i+1])] = self.breaks.temperature[i]
+            p0[(zs>=self.breaks.altitude[i]) & (zs <= self.breaks.altitude[i+1])] = self.breaks.pressure[i]
+            alpha[(zs>=self.breaks.altitude[i]) & (zs <= self.breaks.altitude[i+1])] = -(self.breaks.temperature[i+1] - self.breaks.temperature[i])/ \
+                                                                                        (self.breaks.altitude[i+1] - self.breaks.altitude[i])
+        dz = zs-z0
+        p[alpha == 0.0] = p0[alpha == 0.0]*np.exp(-1*dz[alpha == 0.0]*grav/(gas.gas_specific_constant*T0[alpha == 0.0]))
+        p[alpha != 0.0] = p0[alpha != 0.0]*((1 - alpha[alpha != 0.0]*dz[alpha != 0.0]/T0[alpha != 0.0])**(1*grav/(alpha[alpha != 0.0]*gas.gas_specific_constant)))
+        T = T0 - dz*alpha
+        rho = self.fluid_properties.compute_density(T,p)
+        a = self.fluid_properties.compute_speed_of_sound(T)
+        mew = self.fluid_properties.compute_absolute_viscosity(T)
+        
+        #p, T, rho, a, mew
+        
+        conditions = Data()
+        conditions.freestream = Data()
+        conditions.freestream.pressure = p
+        conditions.freestream.temperature = T
+        conditions.freestream.density = rho
+        conditions.freestream.speed_of_sound = a
+        conditions.freestream.dynamic_viscosity = mew
```

Comment:
I was imagining that in the context of the atmospheric analysis that the data structure would be only one layer deep.  Did you have any strong reasons to write it this way?  We shouldn't add to the data structure's namespace if we don't need to.